docs/gallery/sections/writing_workflows_with_workgraph/autogen/zero_to_hero.py

```diff
@@ -0,0 +1,527 @@
+# because this is a calcfunction, and the input datas are dynamic, we need use **datas.
+def eos(**datas):
+    """Fit the EOS of the data."""
+    from ase.eos import EquationOfState
+
+    #
+    volumes = []
+    energies = []
+    for _, data in datas.items():
+        volumes.append(data.dict.volume)
+        energies.append(data.dict.energy)
+        unit = data.dict.energy_units
+    #
+    eos = EquationOfState(volumes, energies)
+    v0, e0, B = eos.fit()
+    eos = orm.Dict({"unit": unit, "v0": v0, "e0": e0, "B": B})
+    return eos
+
+
+# %%
+# Define the WorkGraph
+# ----------------------
+#
+
+
+from aiida_workgraph import WorkGraph
+
+#
+wg = WorkGraph("eos")
+scale_structure1 = wg.add_task(scale_structure, name="scale_structure1")
+all_scf1 = wg.add_task(
+    all_scf, name="all_scf1", structures=scale_structure1.outputs["structures"]
+)
+eos1 = wg.add_task(eos, name="eos1", datas=all_scf1.outputs["result"])
+wg.to_html()
+#
+
+# %%
+# Combine with a relax task
+# --------------------------
+#
+
+
+from aiida_workgraph import WorkGraph, task
+from aiida_quantumespresso.calculations.pw import PwCalculation
+
+#
+@task.graph_builder(outputs=[{"name": "result", "from": "eos1.result"}])
+def eos_workgraph(structure=None, scales=None, scf_inputs=None):
+    wg = WorkGraph("eos")
+    scale_structure1 = wg.add_task(
+        scale_structure, name="scale_structure1", structure=structure, scales=scales
+    )
+    all_scf1 = wg.add_task(all_scf, name="all_scf1", scf_inputs=scf_inputs)
+    eos1 = wg.add_task(eos, name="eos1")
+    wg.add_link(scale_structure1.outputs["structures"], all_scf1.inputs["structures"])
+    wg.add_link(all_scf1.outputs["result"], eos1.inputs["datas"])
+    return wg
+
+
+#
+
+# -------------------------------------------------------
+wg = WorkGraph("relax_eos")
+relax_task = wg.add_task(PwCalculation, name="relax1")
+eos_wg_task = wg.add_task(
+    eos_workgraph, name="eos1", structure=relax_task.outputs["output_structure"]
+)
+wg.to_html()
+
+
+# %%
+# Useful tool: Web GUI
+# =====================
+# Open a terminal, and run:
+#
+# .. code-block:: console
+#
+#    workgraph web start
+#
+# Then visit the page `http://127.0.0.1:8000/workgraph`, where you can view all the workgraphs.
+#
+# What's Next
+# ===========
+#
+# +-----------------------------------------------------+-----------------------------------------------------------------------------+
+# | `Concepts <../../concept/index.html>`_              | A brief introduction of WorkGraph’s main concepts.                          |
+# +-----------------------------------------------------+-----------------------------------------------------------------------------+
+# | `HowTo <../../howto/index.html>`_                   | Advanced topics, e.g., flow control using `if`, `while`, and `context`.     |
+# +-----------------------------------------------------+-----------------------------------------------------------------------------+
+#
```

Comment:
This is not valid anymore; maybe remote it.
Similar to the WorkChain, we may add some exercises.